### ISSUE TITLE
Chat decides "am I the employer?" from the room's post, not the selected order

### ISSUE THREAD

**Issue by koeylp** [yuthstyle88/108jobs-clean#151]
`ChatRoomView` computes the viewer's role once for the whole conversation:

```ts
const isEmployer = postCreatorId != null && person?.id != null
    ? String(postCreatorId) === String(person?.id)
    : undefined;
```

`postCreatorId` is the creator of the **room's** post. One conversation now
holds every order two people have run together, and the same two people swap
sides between orders, so this is the role for whichever job the room last saw
— not for the order on screen.

Seen live on `dm:8058:8059` (13 orders, 3 posts) signed in as 8058:

- order #10 "VIDEO 0908 — record all core flows" — employer 8058, freelancer
  8059 — the panel says **"You are the freelancer."**
- the room's post is 1305947 ("VERIFY 0909 - roles reversed"), created by 8059,
  which is where that answer comes from.

Consequences beyond the wrong sentence: every employer-only control is gated on
this flag — `onRequestRevision`, `onReleasePayment`, the balance check, and the
"Hire again" button — so on a conversation whose latest job belongs to the
partner, the employer of a finished order cannot act on it at all.

108jobs-flutter fixed the same bug per order (`isEmployerOnSelectedOrder`, PR
#28); the web still reads it from the room. The server already sends
`employerId` / `freelancerId` on every `OrderSummary`, so the fix is to derive
the role from the selected order and fall back to the room only when nothing is
selected.
